Makefile

```diff
@@ -229,6 +230,10 @@ build-plugin-admin-with-oci-discovery: ${CLI_ADMIN_JOBS_OCI_DISCOVERY} publish-a
 .PHONY: build-plugin-admin-with-local-discovery
 build-plugin-admin-with-local-discovery: ${CLI_ADMIN_JOBS_LOCAL_DISCOVERY} publish-admin-plugins-all-local ## Build Tanzu CLI admin plugins with Local standalone discovery
 
+.PHONY: build-plugin-local
+build-plugin-local: ## Build plugin's locally for given plugin name (match-expression) and path
+	tanzu builder cli compile --match "$(PLUGIN_NAME)" --version $(BUILD_VERSION) --ldflags "$(LD_FLAGS)" --path $(PLUGIN_PATH) --target local --artifacts artifacts/${GOHOSTOS}/${GOHOSTARCH}/cli
```

Comment:
1. I don't think we should use `tanzu builder` because the `builder` plugin may not be built, or up to date.
Elsewhere we use `$(GO) run ./cmd/cli/plugin-admin/builder` which will compile and run the `builder` plugin on the spot. Note that the refactoring of the `builder` changes that a little but I can adapt my PR on top of yours: #3349
 
1. I'm a little worried that it is not clear that the user needs to set PLUGIN_NAME.  Without it nothing gets built but it give the impression that it may have built something.  Maybe we need an explicit logic check to handle this case?  I'm not sure.

Reply:
> I'm a little worried that it is not clear that the user needs to set PLUGIN_NAME. Without it nothing gets built but it give the impression that it may have built something. Maybe we need an explicit logic check to handle this case? I'm not sure.
I don't think there is a way to add help text for the make target, as it's being used by developers they must be aware of usage.

Reply:
> I don't think we should use tanzu builder because the builder plugin may not be built, or up to date.
> Elsewhere we use $(GO) run ./cmd/cli/plugin-admin/builder which will compile and run the builder plugin on the spot. Note that the refactoring of the builder changes that a little but I can adapt my PR on top of yours: https://github.com/vmware-tanzu/tanzu-framework/pull/3349

@anujc25 any comments on this?

Reply:
We have added the validation to validate the PLUGIN_NAME, it builds and install only given plugin, if plugin name is empty then it throws an error.